feat: Allow to edit registries

ankanroy-code opened the pull request:
UI - add feature : #156 

make registries editable

https://user-images.githubusercontent.com/58530461/173376189-b06de357-311b-4921-a13c-ac84cbc84e1f.mp4

HOW I DID IT?
basic idea: delete the old registry create a new registry, using the old registry and the changed data.

- create a new component [PreferncesEditRegistrieModal.svelte](https://github.com/ankanroy-code/podman-desktop/blob/main/packages/renderer/src/lib/preferences/PreferncesEditRegistrieModal.svelte)
- used the same logic as [PreferencesRegistriesCreateRegistryModal.svelte](https://github.com/ankanroy-code/podman-desktop/blob/main/packages/renderer/src/lib/preferences/PreferencesRegistriesCreateRegistryModal.svelte)
why did i not used the same component? 
as it was looking more complicated to do! and had to add more rendering logic in the html part.

PROBLEMS?
- only when you change the URL in edit, it create a new registry, and dose not delete the old one.

Please let me know any changes you world like

benoitf commented:
Hello @ankanroy-code and thanks for your contribution 🎉  

To be able to merge pull requests on containers organization, commits need to be signed Off.
https://github.com/containers/podman-desktop/pull/198/checks?check_run_id=6863265316

slemeur commented:
Thanks for your contribution @ankanroy-code !!

benoitf commented:
Hello, you should probbably do a rebase as well of the branch to the main branch.

![image](https://user-images.githubusercontent.com/436777/173381006-014d10e2-a117-48e4-98f2-5bf5597b987e.png)

it looks like there are commits in your PR that are already in the current main branch

ankanroy-code commented:
@benoitf  hey i did a rebase of editmode to my main. did it work. i am new to git. so sorry,it taking time to understand what commands am i passing.

ankanroy-code commented:
@benoitf should i make a new PR from the main?

benoitf commented:
@ankanroy-code changing git history can be done in your branch.  No need to create a new one. But if it's easier for you, you may try with a new one. 
If you're not able to rebase properly I could attempt to do it.

benoitf commented:
@ankanroy-code also probably we would need to share some stuff between the 'creation' and the 'update' component

ankanroy-code commented:
> @ankanroy-code changing git history can be done in your branch. No need to create a new one. But if it's easier for you, you may try with a new one. If you're not able to rebase properly I could attempt to do it.

@benoitf  sorry for late reply, went to college today for the first time. Just came back. If  you so the rebase, please do it, it would be great help.

ankanroy-code commented:
> @ankanroy-code also probably we would need to share some stuff between the 'creation' and the 'update' component

Can you give me some example? Or a bit more specific explanation. I am new, so the terms are not clear. I googled it, but could not understand the question.

benoitf commented:
> Can you give me some example? Or a bit more specific explanation. I am new, so the terms are not clear. I googled it, but could not understand the question.

The idea is that by having a common component, if for example we add a new parameter to the 'creation' panel then it's handled by the 'update/edit' part

benoitf commented:
FYI I did rebase your branch

```
$ git remote add upstream https://github.com/containers/podman-desktop
$ git fetch upstream
$ git pull --rebase upstream main
$ git push -f origin editmode
```

benoitf commented on packages/renderer/src/lib/preferences/PreferencesRegistriesRendering.svelte:
```diff
@@ -3,6 +3,7 @@ import { onMount } from 'svelte';
 import type { Registry } from '@tmpwip/extension-api';
 import { registriesInfos } from '../../stores/registries';
 import PreferencesRegistriesCreateRegistryModal from './PreferencesRegistriesCreateRegistryModal.svelte';
+import PreferncesEditRegistrieModal from './PreferncesEditRegistrieModal.svelte';
```
the filename has a typo (missing e in Preferences) and missing s at `RegistrieS`

ankanroy-code replied:
> FYI I did rebase your branch
> 
> ```
> $ git remote add upstream https://github.com/containers/podman-desktop
> $ git fetch upstream
> $ git pull --rebase upstream main
> $ git push -f origin editmode
> ```

@benoitf  hey I fixed it. Which brunch should I push my code to, Main or editmode.

benoitf replied:
always push to `editmode` (this is a the branch used to create this Pull Request)

ankanroy-code commented:
> FYI I did rebase your branch
> 
> ```
> $ git remote add upstream https://github.com/containers/podman-desktop
> $ git fetch upstream
> $ git pull --rebase upstream main
> $ git push -f origin editmode
> ```

@benoitf  thxxxxxxx

benoitf commented:
There is a PR check that is failing (linter/formatter)

Ensure to run the command `yarn format:fix && yarn lint:fix`

ankanroy-code commented:
@benoitf I fixed the linter/formatter issue

benoitf commented:
@ankanroy-code I don't see the modal file ( I see you removed the old file but in PR diff I don't see the new file)

benoitf commented:
also I don't see any common code between create and edit modal that will lead to something harder to maintain

ankanroy-code commented:
> also I don't see any common code between create and edit modal that will lead to something harder to maintain

Working on. It will be over in 4 hours

ankanroy-code commented:
@benoitf I made a common file for the functions that are reused

benoitf commented:
I've added changes to your branch to use a single modal file (two mode: create or edit). It avoids to duplicate 200lines of code.
Also I've added update event (instead of having to perform remove and then add)

changes are there https://github.com/containers/podman-desktop/pull/198/commits/376cdcf3047637fdbd7cd798f7707a4eb17c01cf

ankanroy-code commented:
> I've added changes to your branch to use a single modal file (two mode: create or edit). It avoids to duplicate 200lines of code. Also I've added update event (instead of having to perform remove and then add)
> 
> changes are there [e458aa5](https://github.com/containers/podman-desktop/commit/e458aa5089fb39ff2fc73597b590f2bc5cffbe08)

@benoitf  Nice, It reduced a lot of code. Also fixed all the problems. Thanks!!! For your help.
